pkg/ride/tree_estimation_test.go

```diff
@@ -406,3 +381,32 @@ func TestBrokenNativeFold(t *testing.T) {
 	_, err = EstimateTree(tree, 4)
 	assert.Error(t, err)
 }
+
+func TestDeclarationDuplication(t *testing.T) {
+	for _, test := range []struct {
+		comment string
+		source  string
+		allowV4 bool
+	}{
+		{`Bad compaction result`, "AAIFAAAAAAAAAA8IAhIAGgkKAmExEgNhMTAAAAALAQAAAAJhMAAAAAAGAQAAAAJhMQAAAAAJAQAAAAJhMAAAAAABAAAAAmEyAAAAAAkBAAAAAmExAAAAAAEAAAACYTMAAAAACQEAAAACYTIAAAAAAQAAAAJhNAAAAAAJAQAAAAJhMwAAAAABAAAAAmE1AAAAAAkBAAAAAmE0AAAAAAEAAAACYTYAAAAACQEAAAACYTUAAAAAAQAAAAJhNwAAAAAJAQAAAAJhNgAAAAABAAAAAmE4AAAAAAkBAAAAAmE3AAAAAAEAAAACYTkAAAAACQEAAAACYTgAAAAAAQAAAAJhMQAAAAAJAQAAAAJhOQAAAAAAAAABAAAAAWkBAAAABGNhbGwAAAAABAAAAAFyCQEAAAACYTEAAAAAAwkAAAAAAAACBQAAAAFyBQAAAAFyBQAAAANuaWwJAAACAAAAAQIAAAAkU3RyaWN0IHZhbHVlIGlzIG5vdCBlcXVhbCB0byBpdHNlbGYuAAAAAChy+pw=", false},
+		{`V5: func f() = {func f() = true; f()}; f()`, "BQoBAAAAAWYAAAAACgEAAAABZgAAAAAGCQEAAAABZgAAAAAJAQAAAAFmAAAAABMDgQM=", true},
+		{`V5: func f() = {func a() = false; 1 + 2}; func g() = {func a() = false; 3 + 5}; f() > g()`, "BQoBAAAAAWYAAAAACgEAAAABYQAAAAAHCQAAZAAAAAIAAAAAAAAAAAEAAAAAAAAAAAIKAQAAAAFnAAAAAAoBAAAAAWEAAAAABwkAAGQAAAACAAAAAAAAAAADAAAAAAAAAAAFCQAAZgAAAAIJAQAAAAFmAAAAAAkBAAAAAWcAAAAAl60UNw==", true},
```

Comment:
I think it would be good idea to add negative test case with correct script.

Reply:
It's already there, the last two is correct scripts that should not rise error. May be `allowV4` is a bad name. Will change it.